agent/api/ecsclient/client.go

```diff
@@ -259,10 +259,37 @@ func getCpuAndMemory() (int64, int64) {
 }
 
 func (client *APIECSClient) getAdditionalAttributes() []*ecs.Attribute {
-	return []*ecs.Attribute{&ecs.Attribute{
+	attributes := []*ecs.Attribute{&ecs.Attribute{
 		Name:  aws.String("ecs.os-type"),
 		Value: aws.String(api.OSType),
 	}}
+	mac, err := client.ec2metadata.PrimaryENIMAC()
+	if err != nil {
+		seelog.Warnf("Unable to get the MAC address of the primary network interface: %v", err)
+		return attributes
+	}
+
+	vpcID, err := client.ec2metadata.VPCID(mac)
+	if err != nil {
+		seelog.Warnf("Unable to get the VPC ID of the primary network interface: %v", err)
+		return attributes
```

Comment:
That's intentional. Though it's debatable if that's optimal. These attributes (at least for now) will be used for determining if an instance satisfies the networking parameters for the backend to create and attach the ENI. I wanted to ensure that either both subnet and vpc id get registered or neither of them do as that's a stronger protocol for the backend to deal with.